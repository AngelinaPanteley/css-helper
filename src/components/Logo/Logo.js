import React from 'react';
import { Link } from 'react-router-dom';

import styles from './Logo.scss';
import logoImg from '../../assets/images/logo.png';

const logo = (props) => {
  return (
    <Link to='/' className={styles.Logo}>
      <img src={logoImg} alt="logo" className={styles.Logo_image} />
    </Link>
  )
}

export default logo;