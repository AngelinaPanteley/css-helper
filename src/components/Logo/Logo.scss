@import '../../variables';

.Logo {
  @include x-center;

  top: 0;

  &_image {
    width: 100px;
  }
}