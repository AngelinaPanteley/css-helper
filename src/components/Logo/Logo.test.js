"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { Link } from 'react-router-dom';
import styles from './Logo.scss';
import logoImg from '../../assets/images/logo.png';
import Logo from './Logo.js';
import { MemoryRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe('>>>Logo', () => {

  //SNAPSHOTS

  const logoComponent = <MemoryRouter><Logo /></MemoryRouter>;
  const component = renderer.create(logoComponent);

  it('renders logo correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <Logo />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders link and image of logo', () => {
    chaiExpect(wrapper.contains(
      <Link to='/' className={styles.Logo}>
        <img src={logoImg} alt="logo" className={styles.Logo_image} />
      </Link>
    )).to.equal(true);
  });

})