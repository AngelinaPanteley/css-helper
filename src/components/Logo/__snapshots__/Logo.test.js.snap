// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Logo renders logo correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <img
    alt="logo"
    src="logo.png"
  />
</a>
`;
